Composer/packages/client/src/components/GetStarted/GetStartedNextSteps.tsx

```diff
@@ -0,0 +1,310 @@
+        highlight: (step) => {
+          setHighlightLUIS(true);
+        },
+        onClick: (step) => {
+          openLink(linkToLUISSettings);
+          if (!step?.checked) {
+            setDisplayManageLuis(true);
+          }
+        },
+      });
+    }
+    if (props.requiresQNA) {
+      newNextSteps.push({
+        key: 'qna',
+        label: formatMessage('Add a QnA Maker key'),
+        description: formatMessage('Your template requires QnA Maker to access content for your bot.'),
+        learnMore: '',
+        required: true,
+        checked: hasQNA,
+        highlight: (step) => {
+          setHighlightQNA(true);
+        },
+        onClick: (step) => {
+          openLink(linktoQNASettings);
+          if (!step?.checked) {
+            setDisplayManageQNA(true);
+          }
+        },
+      });
+    }
+    setRequiredNextSteps(newNextSteps);
+
+    if (props.showTeachingBubble && newNextSteps.length && newNextSteps[0].highlight) {
+      newNextSteps[0].highlight();
+    }
+
+    if (!hasPublishingProfile) {
+      newRecomendedSteps.push({
+        key: 'publishing',
+        label: formatMessage('Create a publishing profile'),
+        description: formatMessage('Set up hosting and other Azure resources to enable publishing'),
+        required: true,
+        checked: hasPublishingProfile,
+        onClick: (step) => {
+          openLink(linkToPublishProfile);
+        },
+      });
+    }
+    setRecommendedNextSteps(newRecomendedSteps);
+
+    const optSteps = [
+      {
+        key: 'packages',
+        label: formatMessage('Add packages'),
+        description: formatMessage('Visit the Package manager to browse packages to add to your bot.'),
+        learnMore: '',
+        checked: false,
+        onClick: () => {
+          openLink(linkToPackageManager);
+        },
+      },
+      {
+        key: 'editlg',
+        label: formatMessage('Edit what your bot says'),
+        description: formatMessage('Customize your bot by editing and adding bot responses.'),
+        learnMore: '',
+        checked: false,
+        onClick: () => {
+          openLink(linkToLGEditor);
+        },
+      },
+      {
+        key: 'editlu',
+        label: formatMessage('Train your language model'),
+        description: formatMessage('Ensure your bot can understand your users by frequently training your LUIS model.'),
+        learnMore: '',
+        checked: false,
+        onClick: () => {
+          openLink(linkToLUEditor);
+        },
+      },
+    ];
+
+    if (hasPublishingProfile) {
+      optSteps.push({
+        key: 'connections',
+        label: formatMessage('Add connections'),
+        description: formatMessage('Connect your bot to Teams, external channels, or enable speech. Learn more'),
+        learnMore: '',
+        checked: false,
+        onClick: () => {
+          openLink(linkToConnections);
+        },
+      });
+    }
+
+    setOptionalSteps(optSteps);
+  }, [botProject]);
```

Comment:
there are a bunch of props being accessed within the effect that may need to be added to the dependency list